npm/src/index.ts

```diff
@@ -106,5 +108,49 @@ export const renderModules = async () => {
     } else {
       errorHandlingModule.handleUploadError(frontEndInfo)
     }
+  } else if (timeoutDialog) {
+    const authModule = await import("./auth")
```

Comment:
There is some complicated logic in this function. Would it be possible to abstract this out into the existing `auth/index.ts` file. Or maybe a new file `àuth/sesssiontimeout.ts`? Can then add some more unit testing around the functionality

Reply:
As suggested, I have moved the session timeout logic to its own file. Still working on the tests though